src/Injector.php

```diff
@@ -150,6 +158,8 @@ public function create(string $name, array $parameters = [])
      * @return object
      * @throws Exception\InvalidCallbackException
      * @throws Exception\ClassNotFoundException
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
```

Comment:
Hm... this is another question if we should include these `@throws` here, as function does not throw these exception explicitly. If these exceptions are thrown in another function call I'm not sure if we should include them here, if so, to what level we should do that?

Reply:
If the exception may happen at runtime, it should be documented here.

Reply:
@webimpress But they are not caught there sow they may finally hit the consumer of this class.
@Ocramius agree

Reply:
@Ocramius I know it would be nice to have it but really hard to accomplish it. I know in PHPStorm you can set it up in the options and define how many nesting level should be checked. We discuss it on the forum already:

https://discourse.zendframework.com/t/php-7-1-and-phpdocs/171

> Document with `@throws` ALWAYS when a method explicitly throws one or more exceptions.
> @weierophinney 

> `@throws` of any sort: if the method throws or rethrows. This may look annoying, but is extremely important to eagerly detect BC breaks related to thrown exception types. A glimpse at the docblock changes in a patch and you know that an added or removed `@throws` is a problem
> @Ocramius